### PR TITLE
Add TREE-QMC package

### PR DESCRIPTION
**Pull request by yhhan19**
Describe your pull request here

TREE-QMC is a quartet-based method for estimating species trees from gene trees. The input/output are the same as the popular method ASTRAL. To learn more about TREE-QMC, check out the related [paper](http://doi.org/10.1101/gr.277629.122) in Genome Research and [abstract](https://link.springer.com/book/10.1007/978-3-031-29119-7) in the RECOMB 2023 proceedings. We wish to add the recipe to build TREE-QMC (https://github.com/molloy-lab/TREE-QMC) to Bioconda.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by yhhan19** [comment]
@BiocondaBot please add label

File: recipes/treeqmc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{% set name = "TREEQMC" %}
```
Comment (thanhleviet) [inline recipes/treeqmc/meta.yaml:1]: Should it be `TREE-QMC` ?

File: recipes/treeqmc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{% set version = "2.0.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/molloy-lab/TREE-QMC/releases/download/2.0.0/TREE-QMC-2.0.0.zip
```
Comment (thanhleviet) [inline recipes/treeqmc/meta.yaml:9]: Can we use the {{ name }} defined in line 1 for the software name in this URL?

File: recipes/treeqmc/meta.yaml (hunk 1 above)
Comment (yhhan19) [reply recipes/treeqmc/meta.yaml:1]: The package name has been updated and now it is tree-qmc.

File: recipes/treeqmc/meta.yaml (hunk 2 above)
Comment (yhhan19) [reply recipes/treeqmc/meta.yaml:9]: Sure.

**Comment by yhhan19** [comment]
@BiocondaBot please add label

**Review by telatin — APPROVED** [review]
well done you all for getting here

**Comment by cbrueffer** [comment]
This PR added both `tree-qmc` and `treeqmc`, which seem identical (minus a different download URL). Was that intentional?

**Pull request merged**